File: platform/Leetcode/src/find_unique_emails.py

def quantify_forwards(emails):
    seen = set()
    for email in emails:
        local, domain = email.split('@')
        if '+' in local:
            local = local[:local.index('+')]
        seen.add(local.replace('.','') + '@' + domain)
    return len(seen)

def convert_to_list(string):
    return [each.strip() for each in string.split(',') if each != '']

def main():
    user_input = input("Enter email addresses, separated by commas : ")
    emails = convert_to_list(user_input)
    output = quantify_forwards(emails)
    print("The unique emails to forward are: ", output)
    
if __name__ == '__main__':
    main()

File: platform/Leetcode/src/findUniqueEmails.java
import java.util.Scanner;
import java.util.HashSet;
import java.util.Set;

public class findUniqueEmails{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter email addresses, separated by commas: ");
        String userInput = sc.nextLine();
        String[] emails = userInput.split(",");
        int numberUnique = quantifyForwards(emails);
        System.out.println("The unique addresses are " + numberUnique);
    }

    public static int quantifyForwards(String[] emails) {
        HashSet<String> seen = new HashSet<String>();
        for (String email : emails) {
            int i = email.indexOf('@');
            String local = email.substring(0, i);
            String rest = email.substring(i);
            if (local.contains("+")) {
                local = local.substring(0, local.indexOf('+'));
            }
            // Note: one should escape the specific character '.',
            // since it is treated as a regex expression.
            local = local.replaceAll("\\.", "");
            seen.add(local + rest);
        }

        return seen.size();
    }

}